be/src/util/metrics.h

```diff
@@ -383,6 +383,15 @@ class MetricRegistry {
     mutable SpinLock _lock;
     std::map<std::string, MetricCollector*> _collectors;
     std::map<std::string, std::function<void()>> _hooks;
+
+    using FlatCollectors = std::vector<std::pair<std::string, MetricCollector>>;
+    using FlatCollectorsPtr = std::shared_ptr<FlatCollectors>;
+    using FlatHooks = std::vector<std::pair<std::string, std::function<void()>>>;
+    using FlatHooksPtr = std::shared_ptr<FlatHooks>;
+
+    // Copy on write
+    std::shared_ptr<FlatCollectors> _all_collectors = std::make_shared<FlatCollectors>();
```

Comment:
I think the `_all_collectors` and `_collectors` both all collectors. the name couldn't reflect the `Copy on write`.

Reply:
I rename _all_collectors to _all_collectors_snapshot